File: apps/api/src/main/java/br/com/agendaplatform/relationships/application/RelationshipSummary.java
package br.com.agendaplatform.relationships.application;

import java.time.Instant;
import java.util.UUID;

public record RelationshipSummary(
        UUID id,
        String name,
        String phone,
        String origin,
        String status,
        Instant lastInteractionAt,
        String nextAction,
        Instant nextActionAt,
        String responsibleName,
        UUID clientId,
        UUID appointmentId,
        boolean pendingContact) {
}

File: apps/admin/src/app/core/relationships/relationship-summary.ts
export type RelationshipStatus =
  | 'NEW_CONTACT'
  | 'IN_SERVICE'
  | 'AWAITING_RESPONSE'
  | 'PENDING_APPOINTMENT'
  | 'SCHEDULED'
  | 'FOLLOW_UP_LATER'
  | 'DID_NOT_SCHEDULE'
  | 'DO_NOT_CONTACT';

export interface RelationshipSummary {
  readonly id: string;
  readonly name: string;
  readonly phone: string;
  readonly origin: string | null;
  readonly status: RelationshipStatus;
  readonly lastInteractionAt: string;
  readonly nextAction: string | null;
  readonly nextActionAt: string | null;
  readonly responsibleName: string;
  readonly clientId: string | null;
  readonly appointmentId: string | null;
  readonly pendingContact: boolean;
}
